mergin/client_pull.py

```diff
@@ -179,10 +179,14 @@ def download_project_is_running(job):
     Returns true/false depending on whether we have some pending downloads.
 
     It also forwards any exceptions from workers (e.g. some network errors). If an exception
-    is raised, it is advised to call download_project_cancel() to abort the job.
+    is raised, aborts the job.
     """
     for future in job.futures:
         if future.done() and future.exception() is not None:
+            # wait for any runinnig futures and cancel any pending futures before performing cleanup, as this may cause
+            # issues while performing cleanup, see https://github.com/MerginMaps/mergin-py-client/issues/157
+            job.is_cancelled = True
+            job.executor.shutdown(wait=True, cancel_futures=True)
```

Comment:
Hmm... this is maybe more tricky...

- shutdown can take quite some time, because we currently do not have a way to cancel futures that are in progress
- this changes the API - the MM plugin would still try to call download_project_cancel()
- didn't know about `cancel_futures=True` - this will be likely useful in download_project_cancel() as well

I will need to think about this a bit more :-)